pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -114,8 +114,21 @@ func (f *accessControlDashboardPermissionFilter) Where() (string, []any) {
 	return f.where.string, f.where.params
 }
 
+func (f *accessControlDashboardPermissionFilter) isEmptyPermissions() bool {
+	permissions := f.user.GetPermissions()
+	if len(permissions) == 1 {
```

Comment:
question: would the issue not occur with service accounts that have some permissions, but none of them are folder/dashboard permissions (eg, service account with permissions on data sources only).

Reply:
Actually, yes, that's a good catch. I've found that the issue occurs when there are no dashboard/folder permissions, so maybe we could extend a check.